model/task/task_annotations.go

```diff
@@ -152,8 +152,12 @@ func UpsertAnnotation(a *annotations.TaskAnnotation, userDisplayName string) err
 	); err != nil {
 		return errors.Wrapf(err, "adding task annotation for task '%s'", a.TaskId)
 	}
-	if a.Issues != nil && len(a.Issues) > 0 {
-		return SetHasAnnotations(a.TaskId, a.TaskExecution)
+	if a.Issues != nil {
```

Comment:
similarly, you don't need to check nil here -- length of the list will just return 0 if its nil

Reply:
ended up having to keep the nil check because it broke the existing tests (that someone previously added)